lib/matplotlib/testing/decorators.py

```diff
@@ -109,18 +110,27 @@ def tearDownClass(cls):
                     cls.original_settings)
 
 
-def cleanup(func):
-    @functools.wraps(func)
-    def wrapped_function(*args, **kwargs):
-        original_units_registry = matplotlib.units.registry.copy()
-        original_settings = mpl.rcParams.copy()
-        try:
-            func(*args, **kwargs)
-        finally:
-            _do_cleanup(original_units_registry,
-                        original_settings)
+def cleanup(style=None):
+    def make_cleanup(func):
+        @functools.wraps(func)
+        def wrapped_function(*args, **kwargs):
+            original_units_registry = matplotlib.units.registry.copy()
+            original_settings = mpl.rcParams.copy()
+            matplotlib.style.use(style)
+            try:
+                func(*args, **kwargs)
+            finally:
+                _do_cleanup(original_units_registry,
+                            original_settings)
 
-    return wrapped_function
+        return wrapped_function
+
+    if isinstance(style, six.string_types):
+        return make_cleanup
+    else:
+        result = make_cleanup(style)
+        style = 'classic'
```

Comment:
No.  The idea here is to keep `@cleanup` without arguments working.  If you pass it no arguments, the first argument should be a function and we want to use classic mode by default.  If you pass a string it specifies a style, and the function will be passed in later (through function chaining).  This is just typical decorator madness, but I can add a comment if you think it would help.